crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -389,7 +390,10 @@ pub type BundledReactions = IndexMap<String, ReactionGroup>;
 
 // The long type after a long visibility specified trips up rustfmt currently.
 // This works around. Report: https://github.com/rust-lang/rustfmt/issues/5703
-type ReactionGroupInner = IndexMap<(Option<OwnedTransactionId>, Option<OwnedEventId>), OwnedUserId>;
+type ReactionGroupInner = IndexMap<
+    (Option<OwnedTransactionId>, Option<OwnedEventId>),
+    (OwnedUserId, MilliSecondsSinceUnixEpoch),
```

Comment:
Can you create named `struct`s for these two tuples? That would make it easier to see what's going on here.

Reply:
I reverted to draft because of the big FIXME, but feedback on the solution I setup for making that usage of a tuple for txn/event ids more explicit will be appreciated @jplatte (e.g. `impl From` seems a bit whacky to me since it's not actual conversions but I'm not sure what would make more sense).